engine/crates/engine/src/error.rs

```diff
@@ -8,18 +8,47 @@ use std::{
 
 use engine_response::error::ErrorExtensionValues;
 use engine_value::ConstValue;
+use runtime::error::PartialErrorCode;
 use thiserror::Error;
 
 pub use engine_response::error::ServerError;
 
 use crate::{LegacyInputType, Pos};
 
-#[derive(Debug, serde::Serialize, serde::Deserialize, strum_macros::Display)]
+#[derive(Clone, Copy, Debug, serde::Serialize, serde::Deserialize, strum_macros::Display)]
 #[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 #[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
 pub enum ErrorCode {
-    PersistedQueryNotFound,
+    BadRequest,
     InternalServerError,
+    TrustedDocumentError,
+    // Used for APQ
+    PersistedQueryError,
+    PersistedQueryNotFound,
+    // Subgraph errors
+    SubgraphError,
+    SubgraphInvalidResponseError,
+    SubgraphRequestError,
+    // Auth
+    Unauthenticated,
+    Unauthorized,
+    // Operation preparation phases
+    OperationParsingError,
+    OperationValidationError,
+    OperationPlanningError,
```

Comment:
Why do engine-v2 errors live inside engine-v1?

Reply:
because the `ErrorCode` was initially created for `PersistedQueryNotFound`. Can create one specific to engine-v2 though if you prefer

Reply:
Not going to insist, but seems like it'd make more sense from an organisational perspective.

Reply:
(Bearing in mind I have thought about this for all of 30 seconds - feel free to disagree)

Reply:
:+1: